hatchery/src/world.rs

```diff
@@ -269,6 +302,32 @@ impl World {
         w.events.push(Event::new(module_id, data));
     }
 
+    fn perform_caller(&self, instance: &Instance, arg_ofs: i32) {
+        let guard = self.0.lock();
+        let w = unsafe { &*guard.get() };
+
+        let caller = w
+            .call_stack
+            .as_ref()
+            .expect("call stack should be set")
+            .caller();
+
+        instance.with_arg_buffer(move |buf| {
```

Comment:
instance has a `let arg_ofs = self.write_to_arg_buffer(arg)?;` that you should use instead of manually writing memory